process/transactionProcessor.go

```diff
@@ -391,27 +394,31 @@ func (tp *TransactionProcessor) GetTransactionStatus(txHash string, sender strin
 	}
 
 	// get status of transaction from random observers
-	tx, err := tp.getTxFromObservers(txHash, requestTypeObservers)
+	tx, err := tp.getTxFromObservers(txHash, requestTypeObservers, false)
 	if err != nil {
 		return UnknownStatusTx, errors.ErrTransactionNotFound
 	}
 
 	return string(tx.Status), nil
 }
 
-func (tp *TransactionProcessor) getTxFromObservers(txHash string, reqType requestType) (*data.FullTransaction, error) {
+func (tp *TransactionProcessor) getTxFromObservers(txHash string, reqType requestType, withEvents bool) (*data.FullTransaction, error) {
```

Comment:
Please split the logic in this function into two functions, such as one of them does not call `alterTxWithScResultsFromSourceIfNeeded` or `mergeScResultsFromSourceAndDestIfNeeded`. This way, it would be trivial to prove (more obvious) that the implementation does never lead to a stackoverflow. 

Currently, `alterTxWithScResultsFromSourceIfNeeded` or `mergeScResultsFromSourceAndDestIfNeeded` contain some risky recursive calls to `getTxFromObserver`.

EDIT: **sorry**, only now I realize that we have `getTxFromObserver` vs. `getTxFromObservers`. I will think of some possible refactoring. If you have any ideas, go on :)